frontend/packages/component-library/CONTRIBUTING.md

```diff
@@ -0,0 +1,190 @@
+# Contributing to the Component Library
+
+Thank you for considering contributing to the Component Library! 🎉  
+We welcome all contributions — whether it’s a bug fix, feature request, documentation improvement, or something else
+entirely.  
+This guide will help you get started quickly and ensure a smooth contribution process. To get full picture of
+the project, please refer to the [README.md](./README.md) file.
+
+## Table of Contents
+
+- [🛣️ Contribution Roadmap](#-contribution-roadmap)
+- [🧑‍💻 How to Add a New Component](#-how-to-add-a-new-component)
+- [🚦 Component Statuses & Production Checklist](#-component-statuses--production-checklist)
+- [🎯 Coding Standards](#-coding-standards)
+- [🧪 How to Test Your Component](#-how-to-test-your-component)
+- [🔀 Commit Message Guidelines](#-commit-message-guidelines)
+- [📚 Need Help?](#-need-help)
+- [💡 Tips for a Smooth PR Review](#-tips-for-a-smooth-pr-review)
+- [✅ How to Get Your PR Approved Faster](#-how-to-get-your-pr-approved-faster)
+
+## 🛣️ Contribution Roadmap
+
+Here’s a quick guide to the contribution process:
+
+1. **(The most important step!) Create a thread** in `#ask-design-system` Slack channel about your change.
+
+   - **Get approval** from @markabarrett and/or @moshebaricdo (design team).
+   - For technical details, ask @levadadenys.
+
+2. **Create a separate PR** with `@code-dot-org/component-library` code changes.
+
+   - Make sure the PR follows the [Best Practices](./README.md/#best-practices).
+   - Include test cases and Storybook updates where applicable.
+
+3. **Get it approved** by:
+
+   - ✅ @moshebaricdo and/or @markabarrett (design)
+   - ✅ @levadadenys (technical implementation)
+
+4. **Merge the PR** once approved.
+
+5. **Use it in your code → Celebrate → Collect tons of appreciation!** 🎉
+
+## 🧑‍💻 How to Add a New Component
+
+1. **Create a new component** in `src/componentLibrary`.
+2. Follow the component structure (see any component for example, for best result - see at least couple components)
+   and [Best Practices](./README.md/#best-practices).
+3. Add a Storybook story under `stories/`.
+4. Write unit tests using Jest + RTL.
+5. Ensure accessibility using `axe` and screen readers.
+6. Submit a PR following the [Contribution Roadmap](#-contribution-roadmap).
+
+## 🚦 Component Statuses & Production Checklist
+
+To give a better understanding of components state/status and which one can and can not be used we've introduced
+a complete `production-ready checklist` and different `statuses` that's applicable to every new and
+existing Design System component.
+
+### Production-ready Checklist:
+
+- Implementation of component approved by design team;
+- Covered with Storybook stories and documentation;
+- Has tests: test every prop, every state and every interaction that's js related
+- Passes accessibility checks (see [README Accessibility Section](./README.md#-accessibility)).;
```

Comment:
I don't think a checklist needs to be done manually, each of which should be done programmatically:

1. Implementation approval by design team: I propose we use CODEOWNERS and have a required approval on design team
2. Covered with storybook stories: We can add a linter rule for this (entrypoint missing `__stories__` directory, for example)
3. Has tests: If we are looking to be strict on this, a branch coverage report that fails CI
4. Passes accessibility checks: We have an automation for this so CI should fail if it passes WCAG automation. Are we looking to do more tests that are manual?

Reply:
It would be nice to automate it, but right now it's not automated, meaning I don't think we should remove it right now. And I'm not sure everything can be automated perfectly (e.g. how to automatically check if stories cover eveything htat we need and follow same naming convention?).
Either way, I think it's worth keeping information in documentation so that people can read more about how everything works.